builtin/stringbuilder_buffer.mbt

```diff
@@ -0,0 +1,70 @@
+// Copyright 2024 International Digital Economy Academy
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+struct StringBuilder {
+  mut bytes : Bytes
+  mut len : Int
+}
+
+pub fn StringBuilder::new(~size_hint : Int = 0) -> StringBuilder {
```

Comment:
@bobzhang - shouldn't the "size_hint" refer to the number of characters and shouldn't the `size_hint = size_hint << 1` be performed within this function and not require users of the `size_hint` to perform the doubling operation themselves since that is an implementation detail?

I ask because of line 76 here:
https://github.com/moonbitlang/core/pull/1731/files#diff-183f011b464e6065c3eef786743e1ba9d2969609afa597570565cf15cd2e9958R76

Either way, it would be nice if there were documentation for `StringBuilder::new` that described whether `size_hint` means the number of `Byte`s or the number of `Char`s.

Reply:
>  shouldn't the "size_hint" refer to the number of characters and shouldn't the size_hint = size_hint << 1 be performed within this function and not require users of the size_hint to perform the doubling operation themselves since that is an implementation detail?

It would mean `Byte`s instead of `Char`s, since the latter is hard to be precise (due to various code ranges)

I will add a docstring later